[#3121] Handle event selection and deselection before firing related events

cadecairos opened the pull request:
https://webmademovies.lighthouseapp.com/projects/65733-popcorn-maker/tickets/3123

secretrobotron commented on src/core/trackevent.js:
```diff
@@ -42,6 +42,7 @@ define( [ "./logger", "./eventmanager", "./observer",
     options = options || {};
 
     var _this = this,
+        _butter = window.Butter.app,
```
Please don't do this; it's not sustainable. Ideally, `butter` shouldn't even be referenced inside this file.

Instead, try employing the observer object to do your bidding. See https://github.com/mozilla/butter/blob/master/src/core/trackevent.js#L142. `notify` works like `dispatch`, except all handlers are called immediately instead of relying on the browser's event loop. So, `src/core/main.js`, you can subscribe to some notification you emit here, keeping `TrackEvent` decoupled from the rest of the code (leading to profit, happiness, and word peace).

Commit:
[#3123] 1. Track Events dispatch trackeventselected/deselected after subscriber callbacks are executed first
2. Selected draggables don't duplicate, stopping that nasty crash when dragging an event
3. Track.deselectEvents now only sets the select state to false if an event is selected, reducing useless calls to all trackevents

mjschranz commented on src/main.js:
```diff
@@ -264,6 +270,10 @@
         var trackEvent = e.data;
+
+        trackEvent.subscribe( "selected", onTrackEventSelected );
+        trackEvent.subscribe( "deselected", onTrackEventDeSelected);
```
Missed a space.

mjschranz commented on src/main.js:
```diff
@@ -272,6 +282,10 @@
             idx = _selectedEvents.indexOf( trackEvent );
+
+        trackEvent.unsubscribe( "selected", onTrackEventSelected );
+        trackEvent.unsubscribe( "deselected", onTrackEventDeSelected);
```
Space!

mjschranz commented on src/core/trackevent.js:
```diff
@@ -344,8 +344,8 @@ define( [ "./logger", "./eventmanager", "./observer",
-       * @event trackeventselected: Dispatched when selected state changes to true.
-       * @event trackeventdeselected: Dispatched when selected state changes to false.
+       * @event trackeventselection: Dispatched when selected state changes to true.
+       * @event trackeventdeselection: Dispatched when selected state changes to false.
```
Why are the event names changed when they aren't what is actually dispatched?

cadecairos replied:
just kruft from an old commit. ill change em
